File: app/src/main/java/com/flybits/conciergejavaexample/Application.java
package com.flybits.conciergejavaexample;

import com.flybits.commons.library.api.FlybitsManager;
import com.flybits.commons.library.api.Region;
import com.flybits.commons.library.logging.VerbosityLevel;
import com.flybits.context.ContextManager;
import com.flybits.context.ReservedContextPlugin;
import com.flybits.context.plugins.FlybitsContextPlugin;

import org.jetbrains.annotations.NotNull;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.provider.Settings;

import java.util.Objects;

public class Application extends android.app.Application {
    static final String CHANNEL_ID = "com.flybits.concierge.channel.id";
    @Override
    public void onCreate() {
        super.onCreate();
        FlybitsManager.setLoggingVerbosity(VerbosityLevel.ALL);

    }

    //TODO start the context plugin
    void startPlugins() {
        // Start location plugin
        ContextManager.start(
                getApplicationContext(),
                new FlybitsContextPlugin.Builder(ReservedContextPlugin.GEOFENCE_LOCATION).build()
        );
        // Start battery plugin
//        ContextManager.start(
//                getApplicationContext(),
//                new FlybitsContextPlugin.Builder(ReservedContextPlugin.BATTERY).build()
//        );
    }
}

File: app/src/main/java/com/flybits/conciergejavaexample/Application.kt
package com.flybits.conciergejavaexample

import android.app.Application
import com.flybits.context.ContextManager.start
import com.flybits.commons.library.api.FlybitsManager
import com.flybits.commons.library.api.idps.AnonymousIDP
import com.flybits.commons.library.api.results.callbacks.BasicResultCallback
import com.flybits.commons.library.exceptions.FlybitsException
import com.flybits.commons.library.logging.VerbosityLevel
import com.flybits.context.ReservedContextPlugin

class Application : Application() {
    lateinit var flybitsManager: FlybitsManager
    override fun onCreate() {
        super.onCreate()
        FlybitsManager.setLoggingVerbosity(VerbosityLevel.ALL)
        flybitsManager = FlybitsManager.Builder(applicationContext)
                .setProjectId("2F0D58EE-B270-4871-8AEF-905226E0088E")
                .setGatewayURL("https://api.development.flybits.com")
                .build()

        flybitsManager.connect(AnonymousIDP(), object : BasicResultCallback {
            override fun onException(exception: FlybitsException) {
                println(" on connect onException")
            }

            override fun onSuccess() {
                println(" on connect onSuccess")

            }

        })
    }

    companion object {
        const val CHANNEL_ID = "com.flybits.concierge.channel.id"
    }
}